### ISSUE TITLE
Panic on text entry interpreted as menu entry

### ISSUE THREAD

**Issue by KillianKemps** [KillianKemps/rs-gopher-client#7]
## How to reproduce bug
Go to `gopher://zaibatsu.circumlunar.space:70/1/updates/`
Then open first entry `1` called "Release Retrospective"

Error: `thread 'main' panicked at 'index out of bounds: the len is 3 but the index is 3', /rustc/4560ea788cb760f0a34127156c78e2552949f734/src/libcore/slice/mod.rs:2717:10`

## Investigation

A `curl gopher://circumlunar.space:70/1/~sloum/phlog` shows that the first entry begins with a `1` so it is normal to be interpreted as a menu entry:
```
curl gopher://circumlunar.space:70/1/~sloum/phlog
i .       .   .     -      ,   -     ,    .	false	null.host	1
i     +              *       _________ .     .	false	null.host	1
i.           *    ,      +  /         \    +	false	null.host	1
i    .    -           .    / //        \ *   .	false	null.host	1
i  ,    .   |   +    '    |             |   .	false	null.host	1
i         --+-- s l o u m | ||          | +	false	null.host	1
i |   *     | *    '   *   \\\___    /\/ .  '	false	null.host	1
i-+- ____     .           . \/   \  /  \  __	false	null.host	1
i |_/    \___     _   +  .  /\/\/\\/  : \/  \	false	null.host	1
i/\\\\o  |\l \___/|\ +    _/ --  |/\: ../  ...	false	null.host	1
i|////o  \\\      L \____/   __ _  _  ___:_ __	false	null.host	1
i               _____________	false	null.host	1
i              |\   phlog.   \  	false	null.host	1
i______________|_\============\_______________ 	false	null.host	1
i	false	null.host	1
1191209-19 - Release Retrospective	/~sloum/phlog/20191209-19.txt	circumlunar.space	70
0191107-22 - Go On Gopher	/~sloum/phlog/20191107-22.txt	circumlunar.space	70
0191018-16 - Jaffar	/~sloum/phlog/20191018-16.txt	circumlunar.space	70
0191005-22 - Bombadillo updates	/~sloum/phlog/20191005-22.txt	circumlunar.space	70
0190719-19 - This and that	/~sloum/phlog/20190719-19.txt	circumlunar.space	70
0190609-16 - Orion	/~sloum/phlog/20190609-16.txt	circumlunar.space	70
[…]
```
However, another Gopher client like VF-1 displays the first item as a text entry when the resource is requested.

It seems the bug is more on the gopherhole side. However, VF-1 manages it in a way to display the menu as a text entry.

Should marmotte fail gracefully like VF-1 into text mode if the menu doesn't display correctly or should marmotte display an error about the requested page without panicking?

**Comment by KillianKemps** [comment]
I correct the first bug description. In fact this entry is a menu entry, but marmotte doesn't parse it well because a tab character is missing here near the end of `gopher://circumlunar.space:70/1/~sloum/phlog/20191209-19.txt`:
```
iThanks to everyone that has taken the time to try out	false	null.host	1
iBombadillo, installed it on their server, submitted issues,	false	null.host	1
ior taken the time to contribute in other way. false	null.host	1
i	false	null.host	1

```

**Comment by KillianKemps** [comment]
vf-1 ignores the error and displays this part like this:
```
Thanks to everyone that has taken the time to try out
Bombadillo, installed it on their server, submitted issues,
or taken the time to contribute in other way. false
```

**Comment by KillianKemps** [comment]
It is finally a similar error than https://github.com/KillianKemps/marmotte/issues/5
But the syntax error in the Gopher menu is different.

**Comment by KillianKemps** [comment]
With the next fix, in #6 this part of the page will be displayed like this:
```
[…]
		Thanks to everyone that has taken the time to try out
		Bombadillo, installed it on their server, submitted issues,
ERR		marmotte: Problem parsing line 112: Could not parse port in: "ior taken the time to contribute in other way. false	null.host	1"
[…]
```

**Comment by KillianKemps** [comment]
PR https://github.com/KillianKemps/marmotte/pull/6 fixes this issue
